store/src/java/com/zimbra/qa/unittest/TestZClient.java

```diff
@@ -255,9 +260,78 @@ public void testImapUIDForNonStandardContact() throws ServiceException {
+        Contact contact = TestUtil.createContactInDefaultFolder(mbox, "testzclient2@example.com");
+        List<ZContact> allZContacts = zmbox.getContactsForFolder(
+                ZFolder.ID_CONTACTS, null /* ids */, (ContactSortBy) null, true /* sync */, null /* attrs */);
+        assertNotNull("zmbox.getAllContacts result should not be null", allZContacts);
+        assertEquals("zmbox.getAllContacts result should have 1 entry", 1, allZContacts.size());
+        ZContact zcontact = allZContacts.get(0);
+        assertEquals("GetAllContacts:Expected=Contact IMAP UID Actual=ZContact IMAP UID",
+                contact.getImapUid(), zcontact.getImapUid());
+        assertEquals("GetAllContacts:Expected=Contact ModifiedSequence Actual=ZContact ModifiedSequence",
+                contact.getModifiedSequence(), zcontact.getModifiedSequence());
+    }
+
+    @Test
+    public void testCreateAndModifyZContact() throws ServiceException {
+        Mailbox mbox = TestUtil.getMailbox(USER_NAME);
+        ZMailbox zmbox = TestUtil.getZMailbox(USER_NAME);
+        Map<String, String> attrs = Maps.newHashMapWithExpectedSize(3);
+        attrs.put(ContactConstants.A_email, "fun@example.com");
+        attrs.put(ContactConstants.A_fullName, "Barney A. Rubble");
+        ZContact zcontact = zmbox.createContact(ZFolder.ID_CONTACTS, null /* tags */, attrs);
+        assertNotNull("Newly created zcontact should not be null", zcontact);
+        assertEquals("Expected=ZContact ID Actual=ZContact IMAP UID", zcontact.getIdInMailbox(), zcontact.getImapUid());
+        Contact contact = mbox.getContactById(null, zcontact.getIdInMailbox());
+        assertNotNull("Contact object gotten by ID on newly created zcontact should not be null", zcontact);
+        assertEquals("IMAP UID Expected=Contact Actual=ZContact", contact.getImapUid(), zcontact.getImapUid());
+        assertEquals("ModifiedSequence Expected=Contact Actual=ZContact",
+                contact.getModifiedSequence(), zcontact.getModifiedSequence());
+        attrs.put(ContactConstants.A_company, "Acme Inc.");
+        zcontact = zmbox.modifyContact(zcontact.getId(), true /* i.e replace all */, attrs);
+        assertNotNull("Contact object from zmbox.modifyContact() should not be null", zcontact);
+        assertEquals("After Modify:Expected=ZContact ID Actual=ZContact IMAP UID",
+                zcontact.getIdInMailbox(), zcontact.getImapUid());
+        contact = mbox.getContactById(null, zcontact.getIdInMailbox());
+        assertNotNull("After Modify:Contact object gotten by ID on newly modified zcontact should not be null",
+                contact);
+        assertEquals("After Modify:IMAP UID:Expected=Contact Actual=ZContact",
+                contact.getImapUid(), zcontact.getImapUid());
+        assertEquals("After Modify:ModifiedSequence Expected=Contact Actual=ZContact",
+                contact.getModifiedSequence(), zcontact.getModifiedSequence());
```

Comment:
I suggest adding a test to check that modseq increased. Ideally, there should be a separate test against Contact (not ZContact) to check that, but I don't see such test anywhere